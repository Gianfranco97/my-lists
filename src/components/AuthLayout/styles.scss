@import "../../colors.scss";

.auth-page-container {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  background-color: $base-color;


  .auth-form {
    width: 400px;
    padding: 50px;
    border-radius: 15px;
    background-color: white;
  }
}
