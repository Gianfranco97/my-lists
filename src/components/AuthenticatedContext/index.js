import React, { useState } from "react";

const AuthenticatedContext = React.createContext();

export function AuthenticatedProvider(props) {
  const [isAuthenticated, setIsAuthenticated] = useState(
    !!sessionStorage.getItem("session-token")
  );

  return (
    <AuthenticatedContext.Provider
      {...props}
      value={{
        isAuthenticated: isAuthenticated,
        changeAuthenticatedStatus: setIsAuthenticated,
      }}
    />
  );
}

export function useAuthenticatedContext() {
  const context = React.useContext(AuthenticatedContext);

  if (!context) {
    throw new Error(
      "'useAuthenticatedContext' must be inside the 'AuthenticatedProvider"
    );
  }

  return context;
}

export default AuthenticatedContext;
