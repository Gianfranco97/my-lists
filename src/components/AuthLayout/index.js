import React from "react";
import PropTypes from "prop-types";

import "./styles.scss";

class AuthLayout extends React.Component {
  render() {
    const { children } = this.props;
    return (
      <div className="auth-page-container">
        <div className="auth-form">
          {children}
        </div>
      </div>
    );
  }
}

AuthLayout.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.array
  ]),
};

export default AuthLayout;
