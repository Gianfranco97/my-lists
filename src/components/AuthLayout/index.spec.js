import React from "react";
import { shallow } from "enzyme";
import AuthLayout from "./index.js";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { AuthenticatedProvider } from "../AuthenticatedContext";

it("The snapshot matches", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <AuthenticatedProvider>
          <AuthLayout>
            <div />
          </AuthLayout>
        </AuthenticatedProvider>
      </MemoryRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("renders without crashing", () => {
  shallow(
    <AuthLayout>
      <div />
    </AuthLayout>
  );
});
