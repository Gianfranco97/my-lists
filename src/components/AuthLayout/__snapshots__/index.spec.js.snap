// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The snapshot matches 1`] = `
<div
  className="auth-page-container"
>
  <div
    className="auth-form"
  >
    <div />
  </div>
</div>
`;
